src/traces/waterfall/cross_trace_calc.js

```diff
@@ -0,0 +1,50 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var Registry = require('../../registry');
+var barCrossTraceCalc = require('../bar/cross_trace_calc').crossTraceCalc;
+
+module.exports = function crossTraceCalc(gd, plotinfo) {
+
+    barCrossTraceCalc(gd, plotinfo);
```

Comment:
Hmm. What happens when we have `bar` and `waterfall` traces on the same subplot?

Reply:
A mock is added to display waterfall with bars. We also have one for waterfall with histogram.
https://github.com/plotly/plotly.js/blob/fed3c97af3997a99e05ddb7fc18a5220d571c235/test/image/baselines/waterfall_and_bar.png